streams/actions/src/main/java/com/hortonworks/streamline/streams/actions/TopologyActions.java

```diff
@@ -39,11 +40,16 @@
     // Compose and run parameter topology as test mode using the underlying streaming engine.
     // The parameter 'topology' should contain its own topology DAG.
     // Please refer the javadoc of TestRunSource and also TestRunSink to see which information this method requires.
-    void testRun(TopologyLayout topology, String mavenArtifacts,
+    void runTest(TopologyLayout topology, TopologyTestRunHistory testRunHistory, String mavenArtifacts,
                  Map<String, TestRunSource> testRunSourcesForEachSource,
                  Map<String, TestRunProcessor> testRunProcessorsForEachProcessor,
                  Map<String, TestRunSink> testRunSinksForEachSink, Optional<Long> durationSecs) throws Exception;
 
+    // kill topology running as test mode if exists
+    // return true if it succeeds to turn on the flag,
```

Comment:
Can you make it clear what you mean by 'flag' and that the implementations are supposed to just submit for killing? Right now can one query the status of killTest via TopologyActions interface ? Maybe this can return a `Future` which can be queried to check the status (can be done later if its more work).

Reply:
Looks like I left a comment about interface too coupled with implement details. I'll leave a requirement of the method.

At least for Storm implementation we don't know whether kill succeeds or not because it just triggers the flag and killing process occurs from another thread.
We still have a chance to determine the status (with some works), but we need to decide whether backend responds immediately and let UI continue polling, or backend waits for killing and responds afterward. If we want former, maybe better not to put effort which we don't actually use.

Reply:
@arunmahadevan 
I just changed description to make requirement of method clearer, but please file an issue if you see the benefit to have return type as `Future`.